# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/chrome-webidl/WindowGlobalActors.webidl
# Commit: 09fcee21b246
# Full Hash: 09fcee21b246dbd3622203d214234ee5116805f4
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-07-10 21:50:49
# Regressor Bug: 1561395
# File Overlap Count: 2
# Description:
#   Bug 1561395 - Move drawSnapshot API to WindowGlobalParent and allow specifying the whole viewport as a rect. r=mattwoodrow,nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D35842
# ==============================================================================

diff -r 0873b1c9bca7 -r 09fcee21b246 dom/chrome-webidl/WindowGlobalActors.webidl
--- a/dom/chrome-webidl/WindowGlobalActors.webidl	Thu Jul 04 00:40:02 2019 +0000
+++ b/dom/chrome-webidl/WindowGlobalActors.webidl	Wed Jul 10 16:45:46 2019 +0000
@@ -57,6 +57,25 @@
     unsigned long long pendingSwitchId);
 
   /**
+   * Renders a region of the frame into an image bitmap.
+   *
+   * @param rect Specify the area of the window to render, in CSS pixels. This
+   * is relative to the current scroll position. If null, the entire viewport
+   * is rendered.
+   * @param scale The scale to render the window at. Use devicePixelRatio
+   * to have comparable rendering to the OS.
+   * @param backgroundColor The background color to use.
+   *
+   * This API can only be used in the parent process, as content processes
+   * cannot access the rendering of out of process iframes. This API works
+   * with remote and local frames.
+   */
+  [Throws]
+  Promise<ImageBitmap> drawSnapshot(DOMRect? rect,
+                                    double scale,
+                                    DOMString backgroundColor);
+
+  /**
    * Fetches the securityInfo object for this window. This function will
    * look for failed and successful channels to find the security info,
    * thus it will work on regular HTTPS pages as well as certificate